ext/drwrap/drwrap.c

```diff
@@ -1122,7 +1120,8 @@ drwrap_set_global_flags(drwrap_global_flags_t flags)
     dr_recurlock_lock(wrap_lock);
     if (dr_atomic_load32(&drwrap_init_count) > 0 &&
         /* After drwrap_init() was called, control inversion cannot be changed. */
-        (flags & DRWRAP_INVERT_CONTROL) != (global_flags & DRWRAP_INVERT_CONTROL)) {
+        TEST(DRWRAP_INVERT_CONTROL, flags) &&
```

Comment:
The original was incorrect, because drwrap_set_global_flags() only adds new flags: it does not replace the set of flags with the passed-in value but just ORs the passed-in value.  (Maybe it should have been called drwrap_add_global_flag() and a separate drwrap_remove_global_flag() added; right now you can never remove a flag.)

Reply:
Added a comment to drwrap_set_global_flags() to clarify this.

Reply:
That check is broken.  Users today, including our own code, make multiple calls to drwrap_set_global_flags() adding one new flag at a time without specifying the other flags in each call.  Nobody is passing the full set they expect to be there, so expecting one particular flag to be in every call is always going to fail over and over.

Reply:
> adding one new flag at a time without specifying the other flags in each call. Nobody is passing the full set they expect to be there,

Oh okay.